From 40c1de0f35fbdd6ab777518a1ed4c0fbf931688e Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Thu, 16 Aug 2012 17:50:47 -0700
Subject: [PATCH 046/974] MIPS: Implement get_cycles() for CONFIG_CEVT_R4K_LIB

Any kernel configured with CEVT_R4K_LIB will have a working
read_c0_count(), so we use it to implement get_cycles().

Signed-off-by: David Daney <david.daney@cavium.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/include/asm/timex.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/mips/include/asm/timex.h b/arch/mips/include/asm/timex.h
index 6529704..83f5504 100644
--- a/arch/mips/include/asm/timex.h
+++ b/arch/mips/include/asm/timex.h
@@ -28,14 +28,18 @@
  * But this only means we'll force a reschedule every 8 seconds or so,
  * which isn't an evil thing.
  *
- * We know that all SMP capable CPUs have cycle counters.
+ * We know that all CONFIG_CEVT_R4K_LIB kernels support cycle counters.
  */
 
 typedef unsigned int cycles_t;
 
 static inline cycles_t get_cycles(void)
 {
+#ifdef CONFIG_CEVT_R4K_LIB
+	return read_c0_count();
+#else
 	return 0;
+#endif
 }
 
 #endif /* __KERNEL__ */
-- 
2.6.2

